Flaky test test_storage_s3/test_invalid_env_credentials.py::test_with_invalid_environment_credentials

Example: https://s3.amazonaws.com/clickhouse-test-reports/73983/3f1181a6a1d390302d0e4dd57cd10cfbd611701d/integration_tests__tsan__[6_6]//home/ubuntu/actions-runner/_work/_temp/test/output_dir/integration_run_parallel4_0.log

Error:
```
==================================== ERRORS ====================================
________ ERROR at teardown of test_with_invalid_environment_credentials ________
[gw2] linux -- Python 3.10.12 /usr/bin/python3

    @pytest.fixture(scope="module")
    def started_cluster():
        try:
            cluster = ClickHouseCluster(__file__)
            cluster.add_instance(
                "s3_with_invalid_environment_credentials",
                with_minio=True,
                env_variables={
                    "AWS_ACCESS_KEY_ID": "aws",
                    "AWS_SECRET_ACCESS_KEY": "aws123",
                },
                main_configs=[
                    "configs/use_environment_credentials.xml",
                    "configs/named_collections.xml",
                ],
                user_configs=["configs/users.xml"],
            )
    
            logging.info("Starting cluster...")
            cluster.start()
            logging.info("Cluster started")
    
            prepare_s3_bucket(cluster)
            logging.info("S3 bucket created")
            run_s3_mocks(cluster)
    
            yield cluster
        finally:
>           cluster.shutdown()

test_storage_s3/test_invalid_env_credentials.py:109: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
helpers/cluster.py:3240: in shutdown
    for line in f:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <encodings.utf_8.IncrementalDecoder object at 0x7f497f642800>
input = b'resolver-1  | \x00\x00UTC0UTC\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x80\x00\x00\x00\x00\xc...c admin update ALIAS` \nminio1-1                                   | \nminio1-1                                   | \n'
final = False

    def decode(self, input, final=False):
        # decode input (taking the buffer into account)
        data = self.buffer + input
>       (result, consumed) = self._buffer_decode(data, self.errors, final)
E       UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 39: invalid start byte

/usr/lib/python3.10/codecs.py:322: UnicodeDecodeError
```

--- a/tests/integration/helpers/cluster.py
+++ b/tests/integration/helpers/cluster.py
@@ -3230,7 +3230,7 @@ class ClickHouseCluster:
 
         if self.up_called:
             # Here errors='replace' because docker can sometimes write non-unicode characters to its output.
-            with open(self.docker_logs_path, "w+", errors='replace') as f:
+            with open(self.docker_logs_path, "w+", errors="replace") as f:
                 try:
                     subprocess.check_call(  # STYLE_CHECK_ALLOW_SUBPROCESS_CHECK_CALL
                         self.base_cmd + ["logs"], stdout=f

--- a/tests/integration/helpers/cluster.py
+++ b/tests/integration/helpers/cluster.py
@@ -3229,7 +3229,8 @@ class ClickHouseCluster:
         fatal_log = None
 
         if self.up_called:
-            with open(self.docker_logs_path, "w+") as f:
+            # Here errors='replace' because docker can sometimes write non-unicode characters to its output.
+            with open(self.docker_logs_path, "w+", errors='replace') as f:
                 try:
                     subprocess.check_call(  # STYLE_CHECK_ALLOW_SUBPROCESS_CHECK_CALL
                         self.base_cmd + ["logs"], stdout=f

